dist/images/Dockerfile.kind

```diff
@@ -0,0 +1,9 @@
+ARG K8S_VERSION="v1.18.2"
+
+FROM kindest/node:$K8S_VERSION
+RUN DEBIAN_FRONTEND=noninteractive clean-install \
+    iputils-ping \
+    tcpdump \
+    vim \
+    openvswitch-switch \
```

Comment:
how do we know what version of OVS we are getting here?

Reply:
is the one that comes with the node image, right now is Ubuntu20-04, but we can put whatever we want, if you link me to a place where we can have a deb package

Reply:
Is it possible to use Fedora as the KIND node image instead of Ubuntu?

Reply:
Yeah, we don't want to add _another_ place where we have to wait for packages to be updated when we need ovn/ovs bugfixes.

If we can't easily have a Fedora-based kind image, then it's fine for CI purposes to use OVS in containers as long as those containers are separate from the OVN containers.

Reply:
fair enough, 
so,  we need to be able to delete an ovnkube-node pod without network disruption, but they are running OVS as a container, so it is not possible righ now.
From your comments I see following options:
1. run OVS in the host (suggested in the original issue and current approach), this has the problem that we can not easily choose the OVS version, so we discarded it
2. run OVS in their own daemon set
3. implement graceful restart in `dist/images/ovnkube.sh` to not flush the flows, but this will block new connections :/
4. ...

Option 2 seems the easiest and cleanest one, right? does it have any problem?

Reply:
> fair enough,
> so, we need to be able to delete an ovnkube-node pod without network disruption, but they are running OVS as a container, so it is not possible righ now.
> From your comments I see following options:
> 
>     1. run OVS in the host (suggested in the original issue and current approach), this has the problem that we can not easily choose the OVS version, so we discarded it
> 
>     2. run OVS in their own daemon set
> 
>     3. implement graceful restart in `dist/images/ovnkube.sh` to not flush the flows, but this will block new connections :/
> 
>     4. ...
> 
> 
> Option 2 seems the easiest and cleanest one, right? does it have any problem?

We already did Option 3 in Openshift for openshift-sdn - basically save and restore existing flows to minimize disruption.
But option 2 seems to work as well - the only downside is when you are going to upgrade the ovs version, you will still have more or less the same issue even if it's a separate daemonset. So am thinking do both 2 and 3 perhaps.

Reply:
I was thinking that this was only used by CI and developers, if that's the case 2 is much simpler :smile:

Reply:
Upstream still needs to solve the upgrade issue - no?

Reply:
No. The recommendation for all real deployments is that they run OVS on the host. For CI, it ends up being simpler to run it in containers, and as long as the e2e tests never touch the OVS containers, this works fine.

Reply:
Ok, in that case, running ovs in a separate daemonset seems to be sufficient.

Reply:
@girishmg we need your input on this one, as you use the templates here in your downstream.

Reply:
PTAL I've moved OVS to its own daemonset